### PR TITLE
feat: add support for custom certificate path

### LINKED ISSUE #1794: [Feature Request]: Support custom certificates for neo4j template
### Related Template(s)

Google_Cloud_to_Neo4j

### What feature(s) are you requesting?

The original issue is tracked at https://issuetracker.google.com/issues/360030796

"""
Currently customers are facing issues establishing a connection from the Dataflow job to a Neo4j database using TLS encryption with certificate validation ("neo4j+s" connection scheme) while utilizing the "Google_Cloud_to_Neo4j" flex template. The connection was successful when using TLS without certificate validation ("neo4j+scc") while with the connection neo4j+s they are experiencing the error 
```
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed:
  sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```

Please advise if there is any way the customers can add "Djavax.net.ssl.trustStore" to JVM when they start a dataflow job without having to build their own custom flex template.
"""

### PR DESCRIPTION
**Pull request by fbiville**
This is an extra attribute that can now be added to the connection JSON information.

Fixes #1794

### CODE REVIEW

File: v2/googlecloud-to-neo4j/src/test/java/com/google/cloud/teleport/v2/neo4j/model/connection/ConnectionParamsTest.java [hunk 1]
```diff
@@ -154,14 +168,15 @@ public void parsesFullCustomInformation() throws Exception {
...
-                "{\"auth_type\": \"custom\", \"server_url\": \"bolt://example.com\", \"database\": \"db\", \"principal\": \"a-principal\", \"credentials\": \"some-credentials\", \"realm\": \"a-realm\", \"scheme\": \"a-scheme\", \"parameters\": {\"foo\": \"bar\", \"baz\": true}}"),
+                "{\"auth_type\": \"custom\", \"server_url\": \"bolt://example.com\", \"database\": \"db\", \"custom_certificate_path\": \"gs://example.com/my/cert\", \"principal\": \"a-principal\", \"credentials\": \"some-credentials\", \"realm\": \"a-realm\", \"scheme\": \"a-scheme\", \"parameters\": {\"foo\": \"bar\", \"baz\": true}}"),
             ConnectionParams.class);
 
     assertThat(connectionParams)
         .isEqualTo(
             new CustomConnectionParams(
                 "bolt://example.com",
                 "db",
+                "gs://example.com/my/cert",
```
Comment (ali-ince) [inline v2/googlecloud-to-neo4j/src/test/java/com/google/cloud/teleport/v2/neo4j/model/connection/ConnectionParamsTest.java:179]: nit: could we also make it clear that this is not `my` but `ca` certificate?

```suggestion
                "gs://example.com/ca/cert",
```

File: v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/model/connection/CustomConnectionParams.java [hunk 2]
```diff
@@ -33,12 +33,13 @@ class CustomConnectionParams extends ConnectionParams {
   public CustomConnectionParams(
       @JsonProperty("server_url") String serverUrl,
       @JsonProperty("database") String database,
+      @JsonProperty("custom_certificate_path") String customCertificatePath,
```
[the remark this answers is not in the document]
Comment (fbiville) [reply v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/model/connection/CustomConnectionParams.java]: fixed in 1242a4af7c76efc8152e02a52ef3250881652d80

File: v2/googlecloud-to-neo4j/src/main/resources/schemas/connection.v1.0.json [hunk 3]
```diff
@@ -17,6 +17,11 @@
       "default": "neo4j",
       "minLength": 1
     },
+    "custom_certificate_path": {
+      "description": "Path of the trusted certificate required to establish connections to Neo4j",
```
[the remark this answers is not in the document]
Comment (fbiville) [reply v2/googlecloud-to-neo4j/src/main/resources/schemas/connection.v1.0.json]: fixed in 1242a4af7c76efc8152e02a52ef3250881652d80

File: v2/googlecloud-to-neo4j/src/test/java/com/google/cloud/teleport/v2/neo4j/model/connection/ConnectionParamsTest.java (hunk 1 above)
Comment (fbiville) [reply v2/googlecloud-to-neo4j/src/test/java/com/google/cloud/teleport/v2/neo4j/model/connection/ConnectionParamsTest.java:179]: fixed in 1242a4af7c76efc8152e02a52ef3250881652d80

**Review by ali-ince — APPROVED** [review]
LGTM

**Comment by fbiville** [comment]
No customer feedback, closing this for now.

**Pull request closed without merging**